pkg/apis/autoscaling/register.go

```diff
@@ -157,4 +157,7 @@ const (
 	// KPALabelKey is the label key attached to a K8s Service to hint to the KPA
 	// which services/endpoints should trigger reconciles.
 	KPALabelKey = GroupName + "/kpa"
+
+	// CheckValidityOnDeployAnnotation
+	CheckValidityOnDeployAnnotation = GroupName + "/checkValidityOnDeploy"
```

Comment:
How about `/scaleToZeroOnDeploy`?

Reply:
I think talking about scaling to zero is probably more accurate since that's what we're really controlling. The fact that scaling to 1 checking the readiness probes is just a side-effect of that. If we go with "scaleToZeroOnDeploy" then the bool is switched :-)  not an issue, just a fun ripple effect